samples/bluetooth/hci_rpmsg/src/main.c

```diff
@@ -230,7 +241,11 @@ static void hci_rpmsg_send(struct net_buf *buf)
 				LOG_WRN("IPC send has been blocked for 1.5 seconds.");
 				retries = 0;
 			}
-			k_yield();
+			if (is_fatal_err) {
+				LOG_ERR("IPC service send error: %d", ret);
+			} else {
+				k_yield();
+			}
```

Comment:
Can you elaborate on why we are not `k_yield`ing when sending fatal errors?

Reply:
In case function `hci_rpmsg_send` is executed in a fatal error context, e.g. by `k_sys_fatal_error_handler` the whole system is in unstable state. There is no point to call `k_yield` in this case because it could make more harm. We do not know if there is any other thread running or if other kernel object are in a valid state. 

I've added a condition here because I didn't want to implement special version of the function for fatal error purposed.

Reply:
Can you add that reasoning as a comment in the code as well?

Reply:
I've added an explanation in the code comment.